pkg/cmd/cli/cmd/debug.go

```diff
@@ -264,6 +276,9 @@ func (o DebugOptions) Validate() error {
 	if len(names) == 0 {
 		return fmt.Errorf("the provided pod must have at least one container")
 	}
+	if (o.AsRoot || o.AsNonRoot) && o.AsUser > 0 {
+		return fmt.Errorf("you may not specify --as-root and --as-user=%d at the same time", o.AsUser)
```

Comment:
Hrm - less to test this way.  Can open it up if someone complains.

On Mon, May 23, 2016 at 9:55 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/debug.go
> <https://github.com/openshift/origin/pull/8929#discussion_r64226267>:
>
> > @@ -264,6 +276,9 @@ func (o DebugOptions) Validate() error {
> >  	if len(names) == 0 {
> >  		return fmt.Errorf("the provided pod must have at least one container")
> >  	}
> > +	if (o.AsRoot || o.AsNonRoot) && o.AsUser > 0 {
> > +		return fmt.Errorf("you may not specify --as-root and --as-user=%d at the same time", o.AsUser)
>
> Shouldn't I be allowed to do --as-root=false --as-user=12? I understand
> the distinction between the options, but I doubt that a user would expect
> the interaction to be this strict.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8929/files/b4053ad3a0bb9872147e534d18dc247aaef25769#r64226267>
>